### PR TITLE
Migrate Add Personal Bank Account flow from useSubStep to useSubPage

### PR DESCRIPTION
**Pull request by VickyStash**
### Explanation of Change

This PR refactors the **Add Personal Bank Account** flow to be URL-driven and cleans up its file structure.

**What this PR does:**

1. **Migrates the flow from `useSubStep` to `useSubPage`.** Each substep is now backed by its own URL segment, so steps are deep-linkable and back navigation works off the route rather than in-memory state. New `SUB_PAGE_NAMES` constants (`plaid`, `manual`, `legal-name`, `address`, `phone-number`, `confirmation`, `success`) are added under `CONST.ADD_PERSONAL_BANK_ACCOUNT`.

2. **Turns the two entry routes into parameterized route objects.** `BANK_ACCOUNT_PERSONAL` and `SETTINGS_ADD_US_BANK_ACCOUNT` change from plain string constants to `{route, getRoute(subPage?, action?)}` objects supporting `:subPage?/:action?` params. Navigation types, `linkingConfig`, and all call sites (`BankAccounts` action, `getPlaidLinkTokenParameters`, `CountrySelectionVerifyAccountPage`, `AccountFlowEntryPoint`, and the corresponding test) are updated accordingly.

3. **Promotes Success to a first-class step.** Instead of the wrapper conditionally rendering a `ConfirmationPage` when `shouldShowSuccess` is set, there's now a dedicated `SuccessStep` substep that the flow advances to once the bank account is added. This also fixes back navigation from the success page and from the Plaid step.

4. **Consolidates the file structure.** The redundant `AddPersonalBankAccountPage.tsx` wrapper is removed, and the flow's component + its `substeps/` and `utils/` are moved from `src/pages/settings/Wallet/InternationalDepositAccount/PersonalInfo/` into a new `src/pages/AddPersonalBankAccountPage/` folder (entry point `index.tsx`). The shared substep imports in `UpdatePersonalBankAccountPage` are repointed to `@pages/AddPersonalBankAccountPage/substeps/...`. The navigator `require` paths resolve to the folder's `index.tsx`, so they need no change.

No user-facing behavior changes are intended beyond the fixed back-navigation edge cases — this is primarily an architecture/refactor PR.

### Fixed Issues

$ https://github.com/Expensify/App/issues/90979
PROPOSAL: N/A

### Tests

This flow is reachable from **two entry points**, both of which must be tested:
- **Wallet** — Settings → Wallet → Add bank account (also via Choose transfer account / Bank account purpose). Uses the `settings/wallet/add-us-bank-account` route.
- **KYC / "get paid"** — the KYC wall (e.g. selecting "Personal bank account" as a payment method when transferring a balance) and the "add a bank account to get paid" CTA on a queued reimbursement in a chat. The chat CTA uses the `bank-account/personal` RHP route and is expected to return you to the originating report on completion.

**A. Wallet entry**
1. From the Wallet settings, start the **Add bank account** flow and choose the personal (US) bank account path.
2. Go through the flow via **Plaid**:
   1. Connect a Plaid account and select an account.
   2. Verify each step (Legal name → Address → Phone number → Confirmation) has its own URL segment (e.g. `.../add-us-bank-account/legal-name`), and that reloading/deep-linking to a step works.
   3. On the Confirmation step, tap a field to edit it, confirm the edit, and verify you are returned directly to the Confirmation step.
   4. Submit on Confirmation and verify the **Success** step is shown, then tap Continue and verify the flow exits back to the Wallet.
3. Repeat the flow via **Manual** entry (Manual bank account details → Legal name → Address → Phone number → Confirmation → Success).

**B. KYC / report entry**
4. From a chat with a queued reimbursement, tap the **add a bank account to get paid** CTA (uses the `bank-account/personal` route).
5. Complete the flow and verify that on success you are returned to the originating report (not the Wallet), i.e. the `exitReportID` path works.
6. Separately, trigger the KYC wall (e.g. transfer balance) and pick **Personal bank account** as the payment method; complete the flow and verify the KYC flow resumes afterwards (the `onSuccessFallbackRoute` / `continueSetup` path).

**C. Back navigation (both entries)**
7. From the Plaid step, press back and verify it returns to the previous screen (not stuck).
8. From the Success step, press back and verify it exits the flow cleanly (to the Wallet or the originating report depending on entry point).
9. From any intermediate step, press back and verify it goes to the previous step.

**D. Shared steps regression**
10. Verify the **Update personal bank account** flow still works (it shares the Address, Legal name, and Phone number steps).

- [ ] Verify that no errors appear in the JS console

### QA Steps

// TODO: These must be filled out, or the issue title must include "[No QA]."

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [ ] I linked the correct issue in the `### Fixed Issues` section above
- [ ] I wrote clear testing steps that cover the changes made in this PR
    - [ ] I added steps for local testing in the `Tests` section
    - [ ] I added steps for the expected offline behavior in the `Offline steps` section
    - [ ] I added steps for Staging and/or Production testing in the `QA steps` section
    - [ ] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [ ] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I ran the tests on **all platforms** & verified they passed on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
- [ ] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [ ] If new assets were added or existing ones were modified, I verified that:
    - [ ] The assets are optimized and compressed (for SVG files, run `npm run compress-svg`)
    - [ ] The assets load correctly across all supported platforms.
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] I added [unit tests](https://github.com/Expensify/App/blob/main/tests/README.md) for any new feature or bug fix in this PR to help automatically prevent regressions in this user flow.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

</details>

### CODE REVIEW

File: src/pages/settings/Wallet/InternationalDepositAccount/PersonalInfo/substeps/SuccessStep.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+                shouldShowButton
+                buttonText={translate('common.continue')}
+                onButtonPress={() => onNext(true)}
+                containerStyle={styles.h100}
+            />
+        </ScrollView>
+    );
+}
+
+SuccessStep.displayName = 'SuccessStep';
```
Comment (VickyStash) [inline src/pages/settings/Wallet/InternationalDepositAccount/PersonalInfo/substeps/SuccessStep.tsx]: Reminder to remove this

**Comment by VickyStash** [comment]
@codex review
